simd/unit_tests/TestSIMD.cpp

```diff
@@ -486,3 +486,62 @@ TEST(simd, device) {
   Kokkos::parallel_for(Kokkos::RangePolicy<Kokkos::IndexType<int>>(0, 1),
                        simd_device_functor());
 }
+
+TEST(simd, test_size) {
```

Comment:
Any good reason to delay assertions to runtime.  These can be done at compile-time via static_assert.

Reply:
Converted to compile time checks. [6f8b0df](https://github.com/kokkos/kokkos/pull/6004/commits/6f8b0dfe179b37f4c9e5a7dc1e7e3e2b636473aa)